### DISCUSSION TITLE
ObservableValidator如何对引用类型里的属性进行校验

### DISCUSSION THREAD

**Discussion by belZxx** [BYJRK/DotNet-Discussions#367] [MVVM]
我有一个继承了ObservableValidator的Model类,里面有多个属性都添加了验证相关的特性.但是当我在ViewModel中使用了这个Model类并使用了 ValidateProperty()对它进行校验后,错误的数据也不会导致HasError的值为true,这种情况应该怎么解决呢?

**Comment by hljlishen** [comment]
是不是给字段写了初始化值导致该字段满足校验条件了呢，比如：
[required]
string name = "111";

**Comment by CodingOctocat** [comment]
数据验证的特性好像要用在属性上，你需要完整属性，并且 `SetProperty` 的第三个参数要填 `true` 启用数据验证。

```cs
private string _name;

[Required]
public string Name
{
    get => _name;
    set => SetProperty(ref _name, value.Trim(), true);
}
```
